ccb/upstream_project.py

```diff
@@ -226,10 +227,58 @@ def __init__(self, recipe):
         super().__init__(recipe, git_url)
         self.owner = owner
         self.repo = repo
+        self.__versions = None
+
+    async def versions(self):
+        if self.__versions is None:
+            try:
+                github_token = get_github_token()
+                headers = {"Accept": "application/vnd.github.v3+json"}
+                if github_token:
+                    headers["Authorization"] = f"token {github_token}"
```

Comment:
I think the CI now fails due to the rate limit of the token (or am I misinterpreting the logs ?)
I'm especially worried as the number of projects can only grow

Reply:
Yes, but it is because I always push 3 branches at the same time: master, dev, and the one of this PR. With each branch triggering 2 workflows, plus the PR's workflow, it makes  a lot more requests than actually needed. The cron workflow always passes on my fork.

Reply:
Mmmm, actually the cron tab also has a problem: https://github.com/ericLemanissier/conan-center-bot/actions/runs/5021405590/jobs/9003755278#step:10:439
I'll see if we can use https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#conditional-requests